checks.d/kafka_consumer.py

```diff
@@ -73,56 +127,75 @@ def check(self, instance):
+        # Report the broker offset
+        # Easiest to do in separate loop from consumer metrics because each
+        # topic/partition combo will have one broker offset, but may have
+        # multiple consumer offsets
+        for (topic, partition), broker_offset in broker_offsets.iteritems():
             broker_tags = ['topic:%s' % topic, 'partition:%s' % partition]
-            broker_offset = broker_offsets.get((topic, partition))
             self.gauge('kafka.broker_offset', broker_offset, tags=broker_tags)
 
-        # Report the consumer
-        for (consumer_group, topic, partition), consumer_offset in consumer_offsets.items():
+        # Report the consumer offset and lag
+        for (consumer_group, topic, partition), consumer_offset in consumer_offsets.iteritems():
+            consumer_group_tags = ['topic:%s' % topic, 'partition:%s' % partition,
+                'consumer_group:%s' % consumer_group]
+            self.gauge('kafka.consumer_offset', consumer_offset, tags=consumer_group_tags)
 
-            # Get the broker offset
             broker_offset = broker_offsets.get((topic, partition))
-
-            # Report the consumer offset and lag
-            tags = ['topic:%s' % topic, 'partition:%s' % partition,
-                    'consumer_group:%s' % consumer_group]
-            self.gauge('kafka.consumer_offset', consumer_offset, tags=tags)
             self.gauge('kafka.consumer_lag', broker_offset - consumer_offset,
-                       tags=tags)
+               tags=consumer_group_tags)
 
     # Private config validation/marshalling functions
 
     def _validate_consumer_groups(self, val):
+        # val = {'consumer_group': {'topic': [0, 1]}}
         try:
-            consumer_group, topic_partitions = val.items()[0]
-            assert isinstance(consumer_group, (str, unicode))
-            topic, partitions = topic_partitions.items()[0]
-            assert isinstance(topic, (str, unicode))
-            assert isinstance(partitions, (list, tuple))
+            # consumer groups are optional
+            assert isinstance(val, dict) or val is None
+            if isinstance(val, dict):
+                for consumer_group, topics in val.iteritems():
+                    assert isinstance(consumer_group, (str, unicode))
+                    # topics are optional
+                    assert isinstance(topics, dict) or topics is None
+                    if isinstance(topics, dict):
+                        for topic, partitions in topics.iteritems():
+                            assert isinstance(topic, (str, unicode))
+                            # partitions are optional
+                            assert isinstance(partitions, (list, tuple)) or partitions is None
+                            if isinstance(partitions, (list, tuple)):
+                                for partition in partitions:
+                                    assert isinstance(partition, int)
```

Comment:
I wish I was just being overly pedantic here. :-)

It actually does need to be an `int` because these partitions are passed directly to `OffsetRequestPayload` in L#135, and if they're passed as strings then the underlying `kafka-python` throws `FailedPayloadsError`. I actually was hitting this for quite a while up in L#98 where partitions are pulled from ZK when they aren't specified in the YAML until I realized what was happening and cast them to `int`. 

I could solve this by just casting to `int` when passing them to `OffsetRequestPayload` in L#135, but that felt more like solving the symptom rather than the root cause. And if this code gets extended later down the road somehow, there's less risk of unexpected side effects if the `int` is forced as early as possible. 

Your thoughts?